### PR TITLE
#433 prompt message before leaving page

### PR DESCRIPTION
**Pull request by asmaNode**
Demo : 
demoinbox#462

### CODE REVIEW

**Review by chamerling — COMMENTED** [review]
From this code, what I understand is that it will prompt every time we want to close the window, not only when some compose are open.

**Comment by asmaNode** [comment]
yes still in progress, i will manage this in the close window service to test if is a composer o rno

**Comment by chamerling** [comment]
Sorry I did not see that was a draft. I was pretty sure that we already have such service available, but I may be wrong.

**Comment by fabienmoyon** [comment]
@chamerling yes here https://github.com/linagora/esn-frontend-mailto/issues/29

Currently mails are save as draft only when we close or sent the mail so we need to improve it.

The goal of this issue is to be able to advise the user before he will lost mail in draft by closing the window. Next step is to add a mail auto save explain here https://github.com/linagora/esn-frontend-inbox/issues/434

File: src/linagora.esn.unifiedinbox/app/services/close-window/close-window.service.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const RETURN_VALUE = 'Are you sure you want to leave?';
```
Comment (chamerling) [inline src/linagora.esn.unifiedinbox/app/services/close-window/close-window.service.js:3]: i18n it. Also explain what will happen if use closes the window

Comment (asmaNode) [reply src/linagora.esn.unifiedinbox/app/services/close-window/close-window.service.js:3]: The close window service will test if the composer is opening or now via getting the status of inboxComposerStatus 
->If it is OPENING the message will be fired 
->If not (DISCARDING)  will exit from function and the message will not be fired

**Comment by asmaNode** [comment]
@chamerling  it is not possible to translate the message it is default by the browser

**Comment by chamerling** [comment]
This is true. But why do we need to use this old alert when we can have our own warning modal? This looks so 2000's

**Comment by asmaNode** [comment]
yes you are right , but it was mentionned in the description https://app.zenhub.com/workspaces/polish-sprint-609a40ec3dc55f000f427b97/issues/linagora/esn-frontend-inbox/433  by @fabienmoyon to follow the mailto SPA method

**Comment by chamerling** [comment]
@asmaNode @fabienmoyon I am not the PO nor BA nor UX, but I feel like having something more modern will be better

**Comment by fabienmoyon** [comment]
@chamerling yes, we will find the best user experience but true solution is do implement an draft auto save and it's [the next story](https://github.com/linagora/esn-frontend-inbox/issues/434)
Like this, this prompt message will be displayed only if the draft auto save was not done before the user what to leave the page

**Review by fabienmoyon — APPROVED** [review]

File: src/linagora.esn.unifiedinbox/app/services/composer-status/composer-status-service.spec.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      const currentStatus = inboxComposerStatus.getStatus();
+
+      expect(currentStatus).to.equal(INBOX_COMPOSER_STATUS.OPENING);
+    });
+
+    it('should neither update the current status nor broadcast an event if the incoming new status is invalid, but log an error', function() {
+      const previousStatus = inboxComposerStatus.getStatus();
+      const newStatus = INBOX_COMPOSER_STATUS.OPENING + 'some_invalid_string_123abc!@#$$^(*)21';
+
+      inboxComposerStatus.updateStatus(INBOX_COMPOSER_STATUS.OPENING + 'some_invalid_string_123abc!@#$$^(*)21');
```
Comment (alagane) [inline src/linagora.esn.unifiedinbox/app/services/composer-status/composer-status-service.spec.js:69]: Is there a reason to not reuse the const `newStatus`?

File: src/linagora.esn.unifiedinbox/app/services/composer-status/composer-status-service.spec.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      inboxComposerStatus.updateStatus(INBOX_COMPOSER_STATUS.OPENING);
+
+      currentStatus = inboxComposerStatus.getStatus();
+
+      expect(currentStatus).to.equal(INBOX_COMPOSER_STATUS.OPENING);
+    });
+  });
+
+  describe('The updateStatus method', function() {
+    it('should update the current status if the incoming new status is valid and broadcast an event with the updated status', function() {
```
Comment (alagane) [inline src/linagora.esn.unifiedinbox/app/services/composer-status/composer-status-service.spec.js:47]: Can the broadcast be tested? (same on next test cases)

Comment (asmaNode) [reply src/linagora.esn.unifiedinbox/app/services/composer-status/composer-status-service.spec.js:47]: i really dont remember why there is a broadcast in the test .I updated the messages in  tests in line with the functionality

**Review by alagane — APPROVED** [review]

**Review by fabienmoyon — APPROVED** [review]

**Pull request merged**